### PR TITLE
add metrics-rds.rb

### PR DESCRIPTION
**Pull request by phoppe93**
its tested and works !

## Pull Request Checklist

**Is this in reference to an existing issue?**
 RDS metrics #137 

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass 

#### New Plugins

- [x] Tests

- [x] Add the plugin to the README

- [x] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

**Comment by phoppe93** [comment]
all checked

**Comment by sstarcher** [comment]
@phoppe93 could you use the aws-sdk v2 instead of v1 we are trying to move away from v1.  Could you also include Common instead of doing the aws logic.  Similar to https://github.com/sensu-plugins/sensu-plugins-aws/blob/master/bin/check-ebs-snapshots.rb

File: bin/metrics-rds.rb [hunk 1]
```diff
@@ -27,57 +27,115 @@
...
+
+  option :accept_nil,
+         short: '-n',
+         long: '--accept_nil',
+         description: 'Continue if CloudWatch provides no metrics for the time period',
+         default: false
+
+ 
 
   def aws_config
```
Comment (sstarcher) [inline bin/metrics-rds.rb]: this block should not be needed

File: bin/metrics-rds.rb [hunk 2]
```diff
@@ -27,57 +27,115 @@
...
-    hash.update access_key_id: config[:access_key_id], secret_access_key: config[:secret_access_key] if config[:access_key_id] && config[:secret_access_key]
-    hash.update region: config[:aws_region]
-    hash
+    { access_key_id: config[:aws_access_key],
+      secret_access_key: config[:aws_secret_access_key],
+      region: config[:aws_region] }
   end
 
-  def run
+  def rds
```
Comment (sstarcher) [inline bin/metrics-rds.rb]: don't pass in aws_config

**Comment by phoppe93** [comment]
i delete the unused block
option :accept_nil,
         short: '-n',
         long: '--accept_nil',
         description: 'Continue if CloudWatch provides no metrics for the time period',
         default: false

i not understand the aws_config issue. pls explain.

**Comment by sstarcher** [comment]
@phoppe93 if you look at the common library you include you will see that it updates the AWS config behind the scenes for you.

File: bin/metrics-rds.rb [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+         default:     :average,
+         proc:        proc { |a| a.downcase.intern },
+         description: 'CloudWatch statistics method'
+
+  def rds
+    @rds ||= config[:role_arn] ? Aws::RDS::Client.new(credentials: role_credentials, region: aws_config[:region]) : Aws::RDS::Client.new(aws_config)
+  end
+
+  def cloud_watch
+    @cloud_watch ||= config[:role_arn] ? Aws::CloudWatch::Client.new(credentials: role_credentials, region: aws_config[:region]) : Aws::CloudWatch::Client.new(aws_config)
```
Comment (sstarcher) [inline bin/metrics-rds.rb]: where does role_arn and role credentials come from?

**Comment by eheydrick** [comment]
@sstarcher are you 👍 on this now?

**Comment by phoppe93** [comment]
@sstarcher when its possible to get the metrics-rds.rb in the master branch ?

**Comment by sstarcher** [comment]
@phoppe93 thanks for the updates!

**Pull request merged**